runhouse/main.py

```diff
@@ -590,159 +593,78 @@ def status(
     _print_status(cluster_status, current_cluster)
 
 
-def get_clusters_from_den():
-    get_clusters_params = {"resource_type": "cluster", "folder": rns_client.username}
-    clusters_in_den_resp = rns_client.session.get(
-        f"{rns_client.api_server_url}/resource",
-        params=get_clusters_params,
-        headers=rns_client.request_headers(),
-    )
-
-    return clusters_in_den_resp
-
-
 @cluster_app.command("list")
-def cluster_list():
-    """List all Runhouse clusters saved in Den"""
-    import sky
+def cluster_list(
+    show_all: bool = typer.Option(
+        False,
+        "--a",
+        "--all",
+        help=f"Get all clusters saved in Den. Up to {MAX_CLUSTERS_DISPLAY} most recently active clusters will be displayed.",
+    ),
+    since: Optional[str] = typer.Option(
+        None,
+        "--since",
+        help="Time duration to filter on. Minimum allowable filter is 1 minute. You may filter by seconds (s), "
+        "minutes (m), hours (h) or days (s). Examples: 30s, 15m, 2h, 3d.",
+    ),
+    cluster_status: Optional[ClustersListStatus] = typer.Option(
+        None,
+        "--status",
+        help="Cluster status to filter on. Supported filter values: running, terminated (cluster is not live), down (Runhouse server is down, but the cluster might be live).",
+    ),
+):
+    """Load Runhouse clusters"""
 
     # logged out case
     if not rh.configs.token:
         # TODO [SB]: adjust msg formatting (coloring etc)
         sky_cli_command_formatted = f"{italic_bold_ansi}sky status -r{reset_format}"  # will be printed bold and italic
         console.print(
-            f"Runhouse token required to view all clusters. Please run `runhouse login` to load your token. To view on-demand clusters run {sky_cli_command_formatted}"
+            f"Listing clusters requires a Runhouse token. Please run  `runhouse login` to get your token, or run {sky_cli_command_formatted} to list locally stored on-demand clusters."
         )
         return
 
-    sky_clusters = sky.status()
+    clusters = Cluster.list(show_all=show_all, since=since, status=cluster_status)
 
-    clusters_in_den_resp = get_clusters_from_den()
+    all_clusters = clusters.get("all_clusters", None)
+    running_clusters = clusters.get("running_clusters", None)
+    sky_clusters = clusters.get("sky_clusters", None)
 
-    if clusters_in_den_resp.status_code != 200:
-        logger.error(
-            f"Failed to load clusters from Den for username: {rns_client.username}"
+    if not all_clusters:
+        additional_info_to_msg = (
+            " that match the provided filters."
+            if (show_all or since or cluster_status)
+            else "."
         )
-        clusters_in_den = []
-    else:
-        clusters_in_den = clusters_in_den_resp.json().get("data")
+        no_clusters_msg = f"No clusters found in Den for {rns_client.username}{additional_info_to_msg}"
```

Comment:
maybe easier to do:

```
no_clusters_msg = f"No clusters found in Den for {rns_client.username}
if show_all or since or cluster_status:
   no_clusters_msg += " that match the provided filters"
```